RabbitMQ.AMQP.Client/Impl/AmqpPublisherBuilder.cs

```diff
@@ -67,8 +67,12 @@ public async Task<IPublisher> BuildAsync(CancellationToken cancellationToken = d
             {
                 address = AddressBuilderHelper.AddressBuilder().Exchange(_exchange).Queue(_queue).Key(_key).Address();
             }
-
-            AmqpPublisher publisher = new(_connection, address, _timeout);
+#if NET6_0_OR_GREATER
+            IMetricsReporter metricsReporter = new MetricsReporter();
+#else
+            IMetricsReporter metricsReporter = new NoOpMetricsReporter();
```

Comment:
by default should be always: `NoOpMetricsReporter`

Reply:
Which conditions should enable metrics ? 
I see in the java implem that it has an AmqpEnvironment that define this option but i do not see this kind of beast in your implem.

Reply:
Lot of dotnet library that i have encounter do not define metrics collection as optionnal (ASpnet , httpclient , Npgsql) or a thing that should be optin. 
Sometimes they provide option to reduce metrics cardinality.

Reply:
@aygalinc, is it okay with me to have metrics enabled by default.  
The important is:
 - does not impact the performances 
 - does not need to include external libraries 

Thank you for working on it

Reply:
It will impact perf because at some point CPU cycle must be used to register metrics if enable by default. 
It will not include additional external library because i use the System.Diagnostic which is already present in the solution.